pipelines/smartseq2_single_sample_unpaired/SmartSeq2SingleSampleUnpaired.wdl

```diff
@@ -0,0 +1,148 @@
+      ref_flat = gene_ref_flat,
+      rrna_intervals = rrna_intervals,
+      output_basename = quality_control_output_basename,
+      stranded = stranded,
+      max_retries = max_retries
+  }
+
+  call Picard.CollectDuplicationMetrics {
+    input:
+      aligned_bam = HISAT2SingleEnd.output_bam,
+      output_basename = quality_control_output_basename,
+      max_retries = max_retries
+  }
+
+  String data_output_basename = output_name + "_rsem"
+  
+  call HISAT2.HISAT2RSEMSingleEnd as HISAT2Transcriptome {
+    input:
+      hisat2_ref = hisat2_ref_trans_index,
+      fastq = fastq,
+      ref_name = hisat2_ref_trans_name,
+      sample_name = sample_name,
+      output_basename = data_output_basename,
+      max_retries = max_retries
+  }
+
+  call RSEM.RSEMExpression {
+    input:
+      trans_aligned_bam = HISAT2Transcriptome.output_bam,
+      rsem_genome = rsem_ref_index,
+      output_basename = data_output_basename,
+      max_retries = max_retries,
+      is_paired = false
+  }
+
+  call GroupQCs.GroupQCOutputs {
+   input:
+      picard_row_outputs = [CollectMultipleMetrics.alignment_summary_metrics,CollectDuplicationMetrics.dedup_metrics,CollectRnaMetrics.rna_metrics,CollectMultipleMetrics.gc_bias_summary_metrics],
+      picard_table_outputs = [CollectMultipleMetrics.base_call_dist_metrics,CollectMultipleMetrics.gc_bias_detail_metrics,CollectMultipleMetrics.pre_adapter_details_metrics,CollectMultipleMetrics.pre_adapter_summary_metrics,CollectMultipleMetrics.bait_bias_detail_metrics,CollectMultipleMetrics.error_summary_metrics],
+      hisat2_stats = HISAT2SingleEnd.log_file,
+      hisat2_trans_stats = HISAT2Transcriptome.log_file,
+      rsem_stats = RSEMExpression.rsem_cnt,
+      output_name = output_name
+  }
+  if (output_zarr) {
+    call ZarrUtils.SmartSeq2ZarrConversion {
+      input:
+        rsem_gene_results = RSEMExpression.rsem_gene,
+        smartseq_qc_files = GroupQCOutputs.group_files,
+        sample_name=sample_name
+    }
+  }
+
+
+  output {
+    # version of this pipeline
+    String pipeline_version = version
+    # quality control outputs
+    File aligned_bam = HISAT2SingleEnd.output_bam
+    File bam_index = HISAT2SingleEnd.bam_index
+    File quality_distribution_metrics = CollectMultipleMetrics.quality_distribution_metrics
+    File quality_by_cycle_metrics = CollectMultipleMetrics.quality_by_cycle_metrics
+    File bait_bias_summary_metrics = CollectMultipleMetrics.bait_bias_summary_metrics
+    File alignment_summary_metrics = CollectMultipleMetrics.alignment_summary_metrics
+    File rna_metrics = CollectRnaMetrics.rna_metrics
+    Array[File] group_results = GroupQCOutputs.group_files
+    # data outputs
+    File aligned_transcriptome_bam = HISAT2Transcriptome.output_bam
+    File rsem_gene_results = RSEMExpression.rsem_gene
+    File rsem_isoform_results = RSEMExpression.rsem_isoform
+
+    # zarr
+    Array[File]? zarr_output_files = SmartSeq2ZarrConversion.zarr_output_files
```

Comment:
I think this could return `Array[File]` if you use `select_all(SmartSeq2ZarrConversion.zarr_output_files)`

Reply:
Didn't like that:
Unable to build WOM node for WorkflowOutput 'zarr_output_files': Cannot build expression for 'SmartSeq2SingleCellUnpaired.zarr_output_files = select_all(SmartSeq2ZarrConversion.zarr_output_files)': select_all failed. It expects an array of optional values but got Array[File]?.